packages/commonwealth/client/scripts/views/pages/Community/Members/GroupsSection/GroupsSection.tsx

```diff
@@ -0,0 +1,145 @@
+    {
+      requirementAmount: '50',
+      requirementChain: 'Ethereum',
+      requirementType: 'Evm Base Tokens',
+      requirementCondition: 'Equal to',
+    },
+  ],
+  topics: [
+    {
+      id: 1,
+      name: 'Call updates',
+    },
+    {
+      id: 2,
+      name: 'Hedgies',
+    },
+  ],
+};
+const sampleGroups = new Array(10)
+  .fill(groupData)
+  .map((x, index) => ({ ...x, isJoined: (index + 1) % 2 === 0 }));
+
+type GroupCategory = 'All groups' | 'In group' | 'Not in group';
+type GroupFilters = {
+  searchText?: string;
+  category?: GroupCategory;
+};
+
+const GroupsSection = () => {
+  const navigate = useNavigate();
+  const [groupFilters, setGroupFilters] = useState<GroupFilters>({
+    searchText: '',
+    category: 'All groups',
+  });
+
+  const getFilteredGroups = () => {
+    return sampleGroups
+      .filter((group) =>
+        groupFilters.searchText
+          ? group.groupName
+              .toLowerCase()
+              .includes(groupFilters.searchText.toLowerCase())
+          : true
+      )
+      .filter((group) =>
+        groupFilters.category === 'All groups'
+          ? true
+          : groupFilters.category === 'In group'
+          ? group.isJoined
+          : !group.isJoined
+      );
+  };
+
+  const navigateToCreateGroupPage = () => {
+    navigate({ pathname: `${app.activeChainId()}/members/groups/create` });
+  };
+
+  return (
+    <section className="GroupsSection">
+      {/* Filter section */}
+      <section
+        className={getClasses<{
+          'cols-3': boolean;
+          'cols-4': boolean;
+        }>(
+          {
+            'cols-3': !Permissions.isCommunityAdmin(),
+            'cols-4': Permissions.isCommunityAdmin(),
+          },
+          'filters'
+        )}
+      >
+        <CWTextInput
+          size="large"
+          fullWidth
+          placeholder="Search groups"
+          iconLeft={<CWIcon iconName="search" className="search-icon" />}
+          onInput={(e) =>
+            setGroupFilters((g) => ({
+              ...g,
+              searchText: e.target.value?.trim(),
+            }))
+          }
+        />
+        <CWText type="b2" fontWeight="bold" className="filter-text">
+          Filter
+        </CWText>
+        <Select
+          containerClassname="select-dropdown"
+          options={groupsFilter.map((x) => ({ id: x, label: x, value: x }))}
+          selected={groupFilters.category}
+          dropdownPosition="bottom-end"
+          onSelect={(item: any) => {
+            setGroupFilters((g) => ({ ...g, category: item.value }));
+          }}
+        />
+        {Permissions.isCommunityAdmin() && (
+          <CWButton
+            buttonWidth="full"
+            label="Create group"
+            iconLeft={'plus'}
+            onClick={navigateToCreateGroupPage}
+          />
+        )}
+      </section>
+
+      {/* Groups list section */}
+      <section className="groups-list">
+        {getFilteredGroups().map((group, index) => (
```

Comment:
'index' is defined but never used

Reply:
Forgot to use it as a `key`, will add that.